tests/src/run.rs

```diff
@@ -313,28 +313,52 @@ impl<'a> Runner<'a> {
     }
 
     /// Display the text for a range.
-    fn text_for_range(&self, range: &Option<Range<usize>>) -> String {
+    fn text_for_range(&self, file: FileId, range: &Option<Range<usize>>) -> String {
         let Some(range) = range else { return "No text".into() };
         if range.is_empty() {
             "(empty)".into()
+        } else if file == self.test.source.id() {
+            format!(
+                "`{}`",
+                self.test.source.text()[range.clone()]
+                    .replace('\n', "\\n")
+                    .replace('\r', "\\r")
+            )
         } else {
-            format!("`{}`", self.test.source.text()[range.clone()].replace('\n', "\\n"))
+            let path = system_path(file).unwrap();
+            let bytes = read(&path).unwrap();
```

Comment:
Why not use `self.world.file(file)` here (and below)? Then we can revert the changes in world.rs.